src/parse_instruction.py

```diff
@@ -237,6 +237,24 @@ class NaiveParsingArch(ArchAsmParsing):
     def normalize_instruction(self, instr: AsmInstruction) -> AsmInstruction:
         return instr
 
+@dataclass
+class UsedRegNames:
+    used_names: Dict[Register, Register] = field(default_factory=dict)
+
+    def input_to_internal(self, reg: Register, arch: ArchAsmParsing) -> Register:
+        if reg in arch.aliased_regs:    
+            if reg not in self.used_names.values():
+                self.used_names[arch.aliased_regs[reg]] = reg
+            return arch.aliased_regs[reg]
+        else:
+            return reg
+
+    def internal_to_output(self, reg: Register) -> Register:
```

Comment:
I think we should keep Registers canonical, i.e. as far as possible avoid having e.g. Register("fv0"). So can you make this return `str` and input_to_internal` accept `str` as input?